examples/applications/plot_cyclical_feature_engineering.py

```diff
@@ -0,0 +1,693 @@
+# synthetic hour data with a bit of extrapolation beyond hour=23:
+hour_df = pd.DataFrame(
+    np.linspace(0, 26, 1000).reshape(-1, 1),
+    columns=["hour"],
+)
+splines = periodic_spline_transformer(24, n_knots=12).fit_transform(hour_df)
+splines_df = pd.DataFrame(
+    splines,
+    columns=[f"spline_{i}" for i in range(splines.shape[1])],
+)
+pd.concat([hour_df, splines_df], axis="columns").plot(x="hour")
+_ = plt.title("Periodic spline-based encoding for the 'hour' feature")
+
+
+# %%
+# Thanks for the use of the `extrapolation="periodic"` parameter, we observe
+# that the feature encoding stays smooth when extrapolating beyond midnight.
+#
+# We can now build a predictive pipeline using this alternative periodic
+# feature engineering strategy.
+#
+# For the "hours" columns we use only 12 knots for a period of 24 hours to
+# avoid over-representing this feature compared to months (12 natural knots)
+# and weekday (7 natural knots).
+cyclic_spline_transformer = ColumnTransformer(
+    transformers=[
+        ("categorical", one_hot_encoder, categorical_columns),
+        ("cyclic_month", periodic_spline_transformer(12), ["month"]),
+        ("cyclic_weekday", periodic_spline_transformer(7), ["weekday"]),
+        ("cyclic_hour", periodic_spline_transformer(24, n_knots=12), ["hour"]),
+    ],
+    remainder="passthrough",
+)
+cyclic_spline_linear_pipeline = make_pipeline(
+    cyclic_spline_transformer,
+    RidgeCV(alphas=alphas),
+)
+evaluate(cyclic_spline_linear_pipeline, X, y, cv=ts_cv)
+
+
+# %%
+# Spline features make it possible for the linear model to successfully
+# leverage the periodic time-related features and reduce the error from ~14% to
+# ~10% of the maximum demand, which is similar to what we observed with the
+# one-hot encoded features.
+#
+# Qualitative analysis of the impact of features on linear models predictions
+# ---------------------------------------------------------------------------
+#
+# Here we want to visualize the impact of the feature engineering choices on
+# the time related-shape of the predictions.
+#
+# To do so we consider an arbitrary time-based split to compare the predictions
+# on a range of held out data points.
+naive_linear_pipeline.fit(X.iloc[train_0], y.iloc[train_0])
+naive_linear_predictions = naive_linear_pipeline.predict(X.iloc[test_0])
+
+one_hot_linear_pipeline.fit(X.iloc[train_0], y.iloc[train_0])
+one_hot_linear_predictions = one_hot_linear_pipeline.predict(X.iloc[test_0])
+
+cyclic_cossin_linear_pipeline.fit(X.iloc[train_0], y.iloc[train_0])
+cyclic_cossin_linear_predictions = cyclic_cossin_linear_pipeline.predict(
+    X.iloc[test_0]
+)
+
+cyclic_spline_linear_pipeline.fit(X.iloc[train_0], y.iloc[train_0])
+cyclic_spline_linear_preictions = cyclic_spline_linear_pipeline.predict(
+    X.iloc[test_0]
+)
+
+# %%
+# We visualize those predictions by zooming on the last 96 hours (4 days) of
+# the test set to get some qualitative insights:
+plt.figure(figsize=(12, 4))
```

Comment:
May we use the OO interface of matplotlib? (especially now that we have so many plots)